Library/Homebrew/os/linux/elf.rb

```diff
@@ -0,0 +1,125 @@
+module ELFShim
+  def elf?
+    return @elf if defined? @elf
+    # See: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+    return @elf = false unless read(4) == "\x7fELF"
+    # Check that the ELF file is for Linux.
+    @elf = case read(1, 7).unpack("C")[0]
+    when 0, 3 then true
+    else false
+    end
+  end
+
+  def arch
+    return :dunno unless elf?
+    @arch ||= case read(2, 18).unpack("v")[0]
+    when 3 then :i386
+    when 62 then :x86_64
+    else :dunno
+    end
+  end
+
+  def elf_type
+    return :dunno unless elf?
+    @elf_type ||= case read(2, 16).unpack("v")[0]
+    when 2 then :executable
+    when 3 then :dylib
+    else :dunno
+    end
+  end
+
+  def dylib?
+    elf_type == :dylib
+  end
+
+  def binary_executable?
+    elf_type == :executable
+  end
+
+  def dynamic_elf?
+    return @dynamic_elf if defined? @dynamic_elf
+    if which "readelf"
+      Utils.popen_read("readelf", "-l", to_path).include?(" DYNAMIC ")
+    elsif which "file"
+      !Utils.popen_read("file", "-L", "-b", to_path)[/dynamic|shared/].nil?
+    else
+      raise StandardError, "Neither `readelf` nor `file` is available "\
```

Comment:
Is `StandardError` not the default for `raise? I can't remember. Also, make this message no more than 80 characters or use `<<~EOS`